buffer/src/main/java/io/netty/buffer/ByteBuf.java

```diff
@@ -1466,6 +1482,24 @@
     public abstract ByteBuf readSlice(int length);
 
     /**
+     * Returns a new retained slice of this buffer's sub-region starting at the current
+     * {@code readerIndex} and increases the {@code readerIndex} by the size
+     * of the new slice (= {@code length}).
+     * <p>
+     * Note that this method returns a {@linkplain #retain() retained} buffer unlike {@link #readSlice(int)}.
+     * This method behaves similarly to {@code readSlice(...).retain()} except that this method may return
+     * a buffer implementation that produces less garbage.
+     *
+     * @param length the size of the new slice
+     *
+     * @return the newly created slice
+     *
+     * @throws IndexOutOfBoundsException
+     *         if {@code length} is greater than {@code this.readableBytes}
+     */
+    public abstract ByteBuf readRSlice(int length);
```

Comment:
to go along with https://github.com/netty/netty/pull/5144#discussion_r59820107 ... `readRSlice` -> `readSliceAndRetain()`.